.github/workflows/ci.yml

```diff
@@ -0,0 +1,247 @@
+name: ci
+
+on:
+  workflow_call:
+    inputs:
+      # enabled by default, required for merge through default GH check
+      test_ci:
+        description: 'Enable Unit tests'
+        default: true
+        required: false
+        type: boolean
+      # disabled by default, but required for merge, there are two GH checks:
+      # - Non-Application Server integration tests
+      # - Application Server integration tests
+      # opt-in with 'ci:agent-integration'
+      agent_integration_tests_ci:
+        description: 'Enable Agent Integration tests'
+        default: false
+        required: false
+        type: boolean
+      # disabled by default, not required for merge
+      # opt-in with 'ci:jdk-compatibility' tag on PR
+      jdk_compatibility_ci:
+        description: 'Enable JDK compatibility tests'
+        default: false
+        required: false
+        type: boolean
+      # disabled by default, not required for merge
+      # opt-in with 'ci:windows' tag on PR
+      windows_ci:
+        description: 'Enable Windows build & tests'
+        default: false
+        required: false
+        type: boolean
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+    paths-ignore:
+      - '**.md'
+      - '**.asciidoc'
+  pull_request:
+    types:
+      - opened
+      - synchronize
+      - reopened
+      - labeled
+      - ready_for_review
+    paths-ignore:
+      - '**.md'
+      - '**.asciidoc'
+
+env:
```

Comment:
I have not looked it at but maybe these env variables can be defined within the composite action? In case they are not used outside of the `uses: ./.github/workflows/build`

By doing so, we can remove a few lines of code